### PR TITLE
[ENHANCEMENT] Linting

### PR DESCRIPTION
**Pull request by j-gynn**
## About The Pull Request

This is a PR to integrate some code cleanup work I've been doing. The goal is to ultimately have (as close as possible to) zero PyLint warnings on every file in the entire `scripts` folder... but that's a looong way away!
### Broad overview of changes
- Consistency with PEP8, using PyLint's default settings.
- Docstrings for functions/methods written in PEP257-compliant reST format, planning to update extant docstrings to follow the same format too.
- Extracting methods from some of the unreasonably long functions (e.g. moving faded-specific initialisation to its own function and calling that during `__init__` if applicable because faded and non-faded cats are fundamentally different).

As it stands, there have been NO file/structural changes. However, I strongly believe some of the code in `cats.py` should not be in that file (looking STRONGLY at `generate_lead_ceremony`, for example). That said, I don't think I can nor should make that call. If there's a way to present a proposal to the senior devs/leads then please let me know and I will write something up.

I will not be touching anything related to moonskip events because I know that is having a major overhaul right now. Please advise if there's anything else that there's no point working on due to upcoming major changes!

Below, you will find an assortment of bugs I've found just by reading the code and logicifying - I will add them as separate bug reports, Soon (tm). 

### Bugs found
- `cats.py - unset_mate` - when breaking up, if `self` and `other_cat` are in a fight, `self_relationship.romantic_love` decrements by 20 points twice (total 40) instead of both it and `other_relationship.romantic_love` decrementing 20 points once (ie 20 each).
- `cats.py - grief` - unused variable `very_high_values` when calculating negative sentiment for moonskip after cat dies.

## Why This Is Good For ClanGen

Code maintenance is essential for well-running projects, especially ones with lots of developers. It helps keep code accessible & understandable to those new to the base; running efficiently for end-users; easy to expand upon for the senior team, etc.

## Linked Issues

Related to a previous PR #2382 , which is why I have made this draft PR. @susieoneil - please feel free to collab on this repo if you want, or alternatively I can merge into yours if you've made significant progress, as I know you said you were going to start tackling the scripts folder in the future. If you can't do that, I can manually give you access - let me know.

## Proof of Testing

No features have been added. Game opens, loads save and runs multiple moonskips with no issues.

## Changelog/Credits

Linting - j-gynn

### CODE REVIEW

**Comment by scribblecrumb** [comment]
Love the look of this!  Honestly a godsend to have someone willing to go through the existing code and clean it up.  I will mention that I've got plans to refactor all the Screens scripts as they're in desperate need of some TLC.  I've not started on that yet though, but if you end up looking those over and have ideas for streamlining then I'll be happy to hear them.

I've also made you an apprentice dev on the discord!  You'll have access to the dev section and we can talk more there :+1:

**Comment by j-gynn** [comment]
I will make a new PR for refactoring work I wanna do, just so we can get the linting done sooner rather than later. One note - I have updated the `.gitignore` - for some reason it had both \ and / variants for some folders and it was throwing errors for black (the formatter this now uses).

**Comment by susieoneil** [comment]
Wow, you have been busy!  Glad my idea was taken on board.

File: scripts/clan_resources/freshkill.py [hunk 1]
```diff
@@ -471,18 +510,30 @@ def feed_group(self, group: list, not_moon_feeding = False, queens = False, fed_
...
                 feeding_amount += 1
-            elif self.total_amount > self.amount_food_needed() and self.nutrition_info[cat.ID].percentage < 100:
+            elif (
+                self.total_amount > self.amount_food_needed()
+                and self.nutrition_info[cat.ID].percentage < 100
+            ):
                 feeding_amount += 0.5
 
-            if not_moon_feeding:
+            if not manual_feeding:
```
Comment (j-gynn) [inline scripts/clan_resources/freshkill.py]: Found the bug! Thank goodness for unit tests.

**Review by j-gynn — COMMENTED** [review]
informational: found the bug

update: there are more.

File: scripts/clan_resources/freshkill.py (hunk 1 above)
Comment (susieoneil) [reply scripts/clan_resources/freshkill.py]: Hooray!

File: scripts/cat/history.py [hunk 2]
```diff
@@ -343,74 +351,73 @@ def add_possible_history(cat, condition:str, death_text:str=None, scar_text:str=
...
         """
 
         History.check_load(cat)
 
         if condition in cat.history.possible_history:
             cat.history.possible_history.pop(condition)
-    
+
     @staticmethod
-    def add_death(cat, death_text, condition=None, other_cat=None, extra_text=None):
```
Comment (j-gynn) [inline scripts/cat/history.py:370]: extra_text does nothing. I removed it but it's being called by death_events to try and add murder_unrevealed_history. I think it should stay removed (and remove it from there too) - and there should be an investigation into whether murder_unrevealed_history is working as intended.

**Comment by j-gynn** [comment]
I think... this might be passing tests now? As an aside, you can actually run tests locally now using the command line and `poetry run pytest` so that's a nice bit of QoL

**Review by scribblecrumb — APPROVED** [review]

File: scripts/cat/history.py (hunk 2 above)
Comment (scribblecrumb) [reply scripts/cat/history.py:370]: Addressed in my moon event refactoring :+1:

**Pull request merged**